controllers/azuremachine_reconciler.go

```diff
@@ -388,19 +370,10 @@ func GetControlPlaneMachines(machineList *clusterv1.MachineList) []*clusterv1.Ma
 	return cpm
 }
 
-func (r *azureMachineReconciler) GetOrCreate() (*compute.VirtualMachine, error) {
-	vm, err := r.findVM()
-	if err != nil {
-		return nil, err
-	}
+// GetRunningVMByTags returns the existing VM or nothing if it doesn't exist.
+func (r *azureMachineReconciler) GetRunningVMByTags(scope *scope.MachineScope) (*infrav1.VM, error) {
```

Comment:
Remove this function as well, it might cause confusion

Reply:
Removed.